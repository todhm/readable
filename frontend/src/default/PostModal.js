import React from 'react';
import {Dialog,DialogActions,DialogContent,Grid,DialogContentText,
    DialogTitle,TextField,Button} from  '@material-ui/core';


export default ({open,handleClose,submitForm,author,title,body}) =>
    <Dialog
      open={open}
      onClose={handleClose}
      aria-labelledby="form-dialog-title"
      fullScreen={true}
      maxWidth="xs"
    >
        <form onSubmit={submitForm}>
          <DialogTitle id="form-dialog-title">Add Your Favorite Post</DialogTitle>
              <DialogContent>
                <TextField
                autoFocus
                margin="dense"
                id="author"
                label="Author"
                name="author"
                type="text"
                value={author}
                />
                <TextField
                  autoFocus
                  margin="dense"
                  id="title"
                  name="title"
                  label="Title"
                  type="text"
                  fullWidth
                  defaultValue={title}
                />
                <TextField
                  autoFocus
                  margin="dense"
                  id="body"
                  name="body"
                  label="Body"
                  type="text"
                  multiline
                  rows="15"
                  fullWidth
                  defaultValue={body}
                />
              </DialogContent>
          <DialogActions>
            <Button onClick={handleClose} color="primary">
              Cancel
            </Button>
            <Button type="submit" color="primary">
              Add Post
            </Button>
          </DialogActions>
      </form>
    </Dialog>
