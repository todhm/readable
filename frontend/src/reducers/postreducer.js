import * as postAction from '../post/actionTypes'

export const initialPost = {
	id: "",
	timestamp: 0,
	title: "",
	body: "",
	author: "",
	category: "",
	voteScore: 1,
	deleted: false,
	votedUp: false,
	votedDown: false,
	commentList: [],
}

const  PostReducer = (state = initialPost, action) => {
	switch (action.type) {
		default: return state
		case postAction.GET_POST:
				return {
				...state,
				...action.data
			}

		case postAction.GET_COMMENTS:
				return {
				...state,
				commentList: action.data
			}

		case postAction.ADD_COMMENTS:
				return {
				...state,
				commentList: [
					...state.commentList,
					action.data
				]
			}
		case postAction.EDIT_COMMENT:
				return {
				...state,
				commentList: [
					...state.commentList.map((comment, index) => {
						if (comment.id !== action.commentId) {
							return comment;
						}
						return {
							...action.data
						};
					})
				]
			}

		case postAction.DELETE_COMMENT:
				return {
				...state,
				commentList: [
					...state.commentList.filter((comment) =>
						comment.id != action.commentId
					)
				]
			}
	}
}

export default PostReducer;
