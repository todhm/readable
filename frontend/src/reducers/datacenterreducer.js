import * as defaultAction from '../default/actionTypes'


const dataCenter = {
	categoryList: {},
	postList: {},

}


const DataCenterReducer = (state = dataCenter, action) => {
	switch (action.type) {
		default: return state

		case defaultAction.GET_CATEGORIES:
				return {
				...state,
				categoryList: action.categoryList,
			}
		case defaultAction.GET_POSTS:
				return {
				...state,
				postList: action.postList,
			}


		case defaultAction.ADD_POST:
				return {
				...state,
				postList: {
					...state.postList,
					[action.payload.id]: {
						...action.payload
					}
				},
			}

		case defaultAction.GET_COMMENTS:
				return {
				...state,
				postList: {
					...state.postList,
					[action.postId]: {
						...state.postList[action.postId],
						commentList: action.payload,
					}
				}
			}

		case defaultAction.UPDATE_POST:
				return {
				...state,
				postList: {
					...state.postList,
					[action.payload.id]: {
						...action.payload
					}
				},
			}

		case defaultAction.DELETE_POST:
				return Object.assign({}, state, {
				postList: Object.keys(state.postList).reduce((result, id) => id === action.postId ?
					result : { ...result,
						[id]: state.postList[id]
					}, {})
			})


		case defaultAction.UPDATE_CATEGORY_SORT:
				return {
				...state,
				categoryList: {
					...state.categoryList,
					[action.category]: {
						...state.categoryList[action.category],
						sortMethod: action.sortMethod
					}
				}
			}
	}

}

export default DataCenterReducer;
