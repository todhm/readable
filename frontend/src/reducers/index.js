import {combineReducers} from 'redux'
import * as defaultAction from '../default/action'
import * as postAction from '../post/action'
const dataCenter={
    categoryList:{},
    postList:{},

}

export const initialPost ={
    id:"",
    timestamp:0,
    title:"",
    body:"",
    author:"",
    category:"",
    voteScore:1,
    deleted:false,
    votedUp:false,
    votedDown:false,
    commentList:[],
}

export const PostReducer =(state=initialPost,action)=>{
    switch(action.type){
        default:
            return state
        case postAction.GET_POST:
            return {
                ...state,
                ...action.data
            }

        case postAction.GET_COMMENTS:
            return {
                ...state,
                commentList:action.data
            }

        case postAction.ADD_COMMENTS:
            return {
                ...state,
                commentList:[
                    ...state.commentList,
                    action.data
                ]
            }
        case postAction.EDIT_COMMENT:
            return {
                ...state,
                commentList:[
                    ...state.commentList.map((comment,index)=>{
                        if(comment.id !== action.commentId) {
                            return comment;
                        }
                        return {
                            ...action.data
                        };
                    })
                ]
            }

        case postAction.DELETE_COMMENT:
            return {
                ...state,
                commentList:[
                    ...state.commentList.filter((comment)=>
                        comment.id!=action.commentId
                    )
                ]
            }
    }
}

const CommentReducer =(state=initialComment,action)=>{
    switch(action.type){
        default:
            return state

    }
}

export const DataCenterReducer =(state=dataCenter,action)=>{
    switch(action.type){
        default:
            return state

        case defaultAction.GET_CATEGORIES:
            return {
                ...state,
                categoryList:action.categoryList,
            }
        case defaultAction.GET_POSTS:
            return {
                ...state,
                postList:action.postList,
            }


        case defaultAction.ADD_POST:
            return {
                ...state,
                postList:{
                    ...state.postList,
                    [action.payload.id]:{
                        ...action.payload
                    }
                },
            }

        case defaultAction.GET_COMMENTS:
            return {
                ...state,
                postList:{
                    ...state.postList,
                    [action.postId]:{
                        ...state.postList[action.postId],
                        commentList:action.payload,
                    }
                }
            }


        case defaultAction.UPDATE_CATEGORY_SORT:
            return {
                ...state,
                categoryList:{
                    ...state.categoryList,
                    [action.category]:{
                        ...state.categoryList[action.category],
                        sortMethod:action.sortMethod
                    }
                }
            }
    }

}


export default combineReducers({
    DataCenterReducer,
    PostReducer
});
