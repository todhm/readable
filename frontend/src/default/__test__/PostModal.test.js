import React from 'react';
import renderer from 'react-test-renderer';
import PostModal from '../PostModal';
import {initialPost} from '../../reducers'
import { mount } from 'enzyme';

const postList=[initialPost]
const sortMethod=""
const mockedEvent = {
      target: {
        value: 'world'
      }
    };
describe('PostModal', () => {
  let component = null;
  const mockHandleClose = jest.fn();
  const mockSubmitForm = jest.fn();
  const mockHandleChange = jest.fn();
  const testAuthor="Lebron Jame";
  const testTitle="How Could I went to Fianl in 2018";
  const testBody="I worked hard";

  it('renders correctly', () => {
    component = mount(
      <PostModal
        open={true}
        handleClose={mockHandleClose}
        submitForm={mockSubmitForm}
        author={testAuthor}
        title={testTitle}
        body={testBody}
        handleChange={mockHandleChange}
        />
    );
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('match text area field', () => {
    expect(component.find('input').at(0).text(), testAuthor);
    expect(component.find('input').at(1).text(), testTitle);
    expect(component.find('textarea').at(0).text(), testBody);
  });

  it('calls functions', () => {
    const buttons = component.find('button');
    buttons.at(0).simulate('click');
    buttons.at(1).simulate('click');

    expect(mockHandleClose.mock.calls.length).toBe(1);
    expect(mockSubmitForm.mock.calls.length).toBe(1);
  });
});
