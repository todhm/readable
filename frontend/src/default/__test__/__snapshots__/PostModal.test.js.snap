// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostModal matches snapshot 1`] = `
<WithStyles(Dialog)
  aria-labelledby="form-dialog-title"
  fullScreen={true}
  maxWidth="xs"
  onClose={[Function]}
  open={true}
>
  <WithStyles(DialogTitle)
    id="form-dialog-title"
  >
    Add Your Favorite Post
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <TextField
      autoFocus={true}
      id="author"
      label="Author"
      margin="dense"
      name="author"
      required={false}
      select={false}
      type="text"
      value="Lebron Jame"
    />
    <TextField
      autoFocus={true}
      fullWidth={true}
      id="title"
      label="Title"
      margin="dense"
      name="title"
      required={false}
      select={false}
      type="text"
      value="How Could I went to Fianl in 2018"
    />
    <TextField
      autoFocus={true}
      fullWidth={true}
      id="body"
      label="Body"
      margin="dense"
      multiline={true}
      name="body"
      required={false}
      rows="15"
      select={false}
      type="text"
      value="I worked hard"
    />
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
      type="submit"
    >
      Add Post
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
